### PR TITLE
Add information on water usage tracking

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds some small docs on tracking water usage.

It seems that tracking water usage is mainly popular in the Netherlands, as I could not find many others.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/energy/water.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+We have the following integrations available for existing products that can provide information about water usage:
+
+- [Flo](/integrations/flo)
+- [Flume](/integrations/flume)
+- [HomeWizard Energy](/integrations/homewizard)
+
+If you have a smart water meter that is wirelessly connected to your electricity meter, the following integration could be of help:
+
+- [P1 Monitor](/integrations/p1_monitor)
```
Comment (klaasnicolaas) [inline source/_docs/energy/water.markdown]: P1 Monitor uses a simple pulse meter coupled to GPIO pins, so I think it falls more into the DIY category than something ready-made and wireless?

Comment (balloob) [reply source/_docs/energy/water.markdown]: Klaas is correct. Docs mention soldering:

https://www.ztatz.nl/p1-monitor-watermeter/

**Review by balloob — DISMISSED** [review]

File: source/_docs/energy/water.markdown (hunk 1 above)
Comment (frenck) [reply source/_docs/energy/water.markdown]: I've removed the section. Added the P1 Monitor to the DIY paragraph and removed Toon. Toon is almost not used and would only work for a very very small percentage of Dutch users that have a smart water meter (which isn't rolled out anymore at this point).

File: source/_docs/energy/water.markdown [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- [HomeWizard Energy](/integrations/homewizard)
+
+Alternatively, the following shops sell ESPHome-based devices, that use a proximity sensor to detect a rotating magnet in your water meter and use that pulse to count each liter of water used.
+
+- [S0tool](https://huizebruin.github.io/s0tool/) ("Made for ESPHome" approved)
+- [Waterlezer dongle](https://smart-stuff.nl/product/esphome-waterlezer-dongle/) (Dutch)
+- [Slimme Watermeter Gateway](https://smartgateways.nl/product/slimme-watermeter-gateway/) (Dutch)
+- [watermeterkit.nl](https://watermeterkit.nl/) (Dutch)
+
+Maybe you like to build one yourself? Pieter Brinkman has quite a [nice blog article on how to creator your own water sensor](https://www.pieterbrinkman.com/2022/02/02/build-a-cheap-water-usage-sensor-using-esphome-home-assistant-and-a-proximity-sensor/) using ESPHome, or [built a water meter](https://www.ztatz.nl/p1-monitor-watermeter/) that works with the [P1 Monitor](/integrations/p1_monitor) integration.
```
Comment (MartinHjelmare) [inline source/_docs/energy/water.markdown]:
```suggestion
Maybe you like to build one yourself? Pieter Brinkman has quite a [nice blog article on how to create your own water sensor](https://www.pieterbrinkman.com/2022/02/02/build-a-cheap-water-usage-sensor-using-esphome-home-assistant-and-a-proximity-sensor/) using ESPHome, or [build a water meter](https://www.ztatz.nl/p1-monitor-watermeter/) that works with the [P1 Monitor](/integrations/p1_monitor) integration.
```

**Review by pvizeli — DISMISSED** [review]

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**